Added blogpost and sample view for cron runtime monitoring

gagan93jtg opened the pull request.

Commit:
Added blogpost and sample view for cron runtime monitoring

Commit:
Add authors. Minor changes in post

Commit:
Grammar corrected. Updated bar to color

ShubhamGupta commented on _posts/2019-05-03-monitoring-cron-runtime-in-rails.md:
```diff
@@ -0,0 +1,157 @@
+---
+layout: post
+title: Monitoring Cron Runtime In Ruby
+categories: [ruby, whenever, redis, rake, active_admin]
```
Lets keep categories very focussed. For example: This is not really a post on Redis.

ShubhamGupta commented on _posts/2019-05-03-monitoring-cron-runtime-in-rails.md:
```diff
@@ -0,0 +1,157 @@
+author: gagan93
+---
+
+We have our web backend in Ruby on rails, and we use many gems for different purposes. One such useful gem is **[whenever](https://github.com/javan/whenever)**. We have hundreds of cronjobs running per day. As our system scaled, it was essential to make sure that cronjobs run in the expected time. For, e.g., if a cronjob has to run every hour, we must make sure that one copy must run in less than ...
```
1. Lets make hundreds of cronjobs to "thousands".
2. Example can be more clear. One use case is -- There should not be more than one occurrence running at a time. Other case is that if expected runtime is 5 min, then we should get notified if it exceeds. This second thing is not discussed.
3. We should add a one-liner for what whenever does. Something like "We use whenever to manage these cron tasks."

gagan93jtg replied:
1. Sure
2. Mentioned multiple copies case also. Whenever supports that already.
3. It is explained in first para, a little later

> Though whenever provided a very clean syntax for defining and deploying cronjobs, there was no insight on what’s going on

>

ShubhamGupta commented on _posts/2019-05-03-monitoring-cron-runtime-in-rails.md:
```diff
@@ -0,0 +1,157 @@
+# Rakefile
+tasks = Rake.application.tasks
+tasks.each do |task|
+  next unless task.to_s.starts_with?('loconav_crons:')
```
Lets not use `loconav` here.

ShubhamGupta commented on _posts/2019-05-03-monitoring-cron-runtime-in-rails.md:
```diff
@@ -0,0 +1,157 @@
+
+  if constraints[:run_at_ist]
+    constraints[:run_at_ist] = [constraints[:run_at_ist]] unless constraints[:run_at_ist].is_a? Array
+    run_at_utc = constraints[:run_at_ist].map{ |run_at_ist| (Time.parse(run_at_ist) - 19800).strftime('%H:%M %P') }
```
Remove IST related info.

Commit:
Update post tags as discussed.
Minor content changes as discussed.
Remove unreqd code.
Add sample code for redis wrapper and mailer.

Commit:
Minor grammar changes

ShubhamGupta committed:
Minor grammatical fixes.

Commit:
Minor italics change
